electricitymap/contrib/lib/models/event_lists.py

```diff
@@ -57,8 +99,39 @@ def append(
         if event:
             self.events.append(event)
 
+    @staticmethod
+    def merge_exchanges(
+        ungrouped_exchanges: List["ExchangeList"], logger: Logger
+    ) -> "ExchangeList":
+        """
+        Given multiple parser outputs, sum the netflows of corresponding datetimes
+        to create a unique exchange list. Sources will be aggregated in a
+        comma-separated string. Ex: "entsoe, eia".
+        """
+        exchanges = ExchangeList(logger)
+        if ExchangeList.is_completly_empty(ungrouped_exchanges, logger):
+            return exchanges
+
+        # Create a dataframe for each parser output, then flatten the exchanges.
+        exchange_dfs = [
+            pd.json_normalize(exchanges.to_list()).set_index("datetime")
+            for exchanges in ungrouped_exchanges
+            if len(exchanges.events) > 0
+        ]
+
+        exchange_df = pd.concat(exchange_dfs)
+        exchange_df = exchange_df.rename(columns={"sortedZoneKeys": "zoneKey"})
+        zone_key, sources, source_type = ExchangeList.get_unique_zone_source(
+            exchange_df
+        )
+        exchange_df = exchange_df.groupby(level=0, dropna=False).sum()
```

Comment:
I assume this will work if there none/null events in on of the exchange lists, but should we not log this so we know if we are potentially missing data and should try and re-fetch it later?

Reply:
This will be logged when the list you want to merge is created in the first place.